src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -864,7 +864,7 @@ public void finishTree(DetailAST rootAST) {
     /** Examine the order of all the imports and log any violations. */
     private void finishImportList() {
         String currentGroup = getFirstGroup();
-        int currentGroupNumber = customOrderRules.indexOf(currentGroup);
+        int currentGroupNumber = customOrderRules.lastIndexOf(currentGroup);
```

Comment:
This is not a nano-optimization, this is a hack to avoid a false-positive: invocation of the method `lastIndexOf` marks a collection is a `List`, while `indexOf` [doesn't](https://github.com/mebigfatguy/fb-contrib/blob/28c23d607aa5d4eee2982cf5c0ccbbce491db8db/src/main/java/com/mebigfatguy/fbcontrib/detect/DubiousListCollection.java#L77).

Reply:
may be report a bug?

Reply:
If I got the comment message correctly, this behavior is intended.